Revert "Remove `test_storage_rabbitmq`"

nikitamikhaylov opened the pull request:
Reverts ClickHouse/ClickHouse#92127. 

Analyzing [latest failures](https://play.clickhouse.com/play?user=play#V0lUSAogICAgMzAgQVMgbGFzdF9kYXlzX3RvX2NoZWNrLAogICAgMTAwIEFTIG1heF90ZXN0c190b19zaG93LAogICAgNSBhcyBtYXhfcHJzX3RvX3Nob3csCiAgICA3MCBBUyB0ZXN0X25hbWVfbWF4X2xlbmd0aApTRUxFQ1QKICAgIGNvdW50KCksCiAgICByZXBvcnRfdXJsCkZST00gY2hlY2tzCldIRVJFCiAgICBjaGVja19zdGFydF90aW1lID49IG5vdygpIC0gaW50ZXJ2YWwgbGFzdF9kYXlzX3RvX2NoZWNrIGRheQogICAgQU5EICh0ZXN0X3N0YXR1cyA9ICdGQUlMJykKICAgIEFORCB0ZXN0X25hbWUgSUxJS0UgJyV0ZXN0X3N0b3JhZ2VfcmFiYml0bXElJwpHUk9VUCBCWSByZXBvcnRfdXJsCk9SREVSIEJZIGNvdW50KCkgREVTQwpMSU1JVCBtYXhfdGVzdHNfdG9fc2hvdw==), here's the list of actions taken:

- Update rabbitmq from 4.0.4 to latest 4.2.1
- Increase timeouts to prevent pytest killing the test before pulling the docker image: https://github.com/ClickHouse/ClickHouse/pull/92151/commits/d6d98178b14d27e41dd9ee707712a136e29831d3. Addresses issues like [this](https://s3.amazonaws.com/clickhouse-test-reports/json.html?REF=25.9&sha=7737bba9345896fee0e3805ee54bf8d8b51d34d7&name_0=ReleaseBranchCI&name_1=Integration%20tests%20%28amd_asan%2C%20old%20analyzer%2C%205%2F6%29&name_1=Integration%20tests%20%28amd_asan%2C%20old%20analyzer%2C%205%2F6%29) and [this](https://s3.amazonaws.com/clickhouse-test-reports/json.html?REF=25.8&sha=5aaf6344ab535b81ca27b3a2aaccedb85e735d05&name_0=ReleaseBranchCI&name_1=Integration%20tests%20%28amd_asan%2C%20old%20analyzer%2C%201%2F6%29&name_1=Integration%20tests%20%28amd_asan%2C%20old%20analyzer%2C%201%2F6%29)
- Refactor reset_rabbitmq to stop and start a new docker instance from scratch to avoid stop_app/start_app/reset timeouts such [this](https://s3.amazonaws.com/clickhouse-test-reports/PRs/90016/08602988fc37fef9867149ca3cc13ed97f81baf6//integration_tests_arm_binary_distributed_plan_4_4/job.log) and [this](https://s3.amazonaws.com/clickhouse-test-reports/REFs/master/660e61056378847a8cfd567d86e3fa120f7edcc7//integration_tests_arm_binary_distributed_plan_4_4/job.log)
- Remove RabbitMQMonitor and all its complexity and use the management plugin to collect stats after every test. They look like this
```
2025-12-19 07:17:42.059000 [ 528 ] DEBUG : RabbitMQ statistics:
{
  "management_version": "4.0.4",
  "rates_mode": "basic",
  "sample_retention_policies": {
    "global": [
      600,
      3600,
      28800,
      86400
    ],
    "basic": [
      600,
      3600
    ],
    "detailed": [
      600
    ]
  },
  "exchange_types": [
    {
      "name": "direct",
      "description": "AMQP direct exchange, as per the AMQP specification",
      "enabled": true
    },
    {
      "name": "fanout",
      "description": "AMQP fanout exchange, as per the AMQP specification",
      "enabled": true
    },
    {
      "name": "headers",
      "description": "AMQP headers exchange, as per the AMQP specification",
      "enabled": true
    },
    {
      "name": "topic",
      "description": "AMQP topic exchange, as per the AMQP specification",
      "enabled": true
    },
    {
      "name": "x-consistent-hash",
      "description": "Consistent Hashing Exchange",
      "enabled": true
    },
    {
      "name": "x-local-random",
      "description": "Picks one random local binding (queue) to route via (to).",
      "enabled": true
    }
  ],
  "product_version": "4.0.4",
  "product_name": "RabbitMQ",
  "rabbitmq_version": "4.0.4",
  "cluster_name": "rabbit@rabbitmq1",
  "cluster_tags": [],
  "node_tags": [],
  "erlang_version": "27.2",
  "erlang_full_version": "Erlang/OTP 27 [erts-15.2] [source] [64-bit] [smp:32:32] [ds:32:32:10] [async-threads:1] [jit:ns]",
  "release_series_support_status": "supported",
  "disable_stats": true,
  "is_op_policy_updating_enabled": true,
  "enable_queue_totals": false,
  "node": "rabbit@rabbitmq1",
  "object_totals": {
    "queues": 0,
    "exchanges": 8,
    "connections": 0
  }
}
```

nikitamikhaylov committed:
Revert "Remove `test_storage_rabbitmq`"

pamarcos committed:
Merge branch 'master' into revert-92127-remove-test-rabbitmq

pamarcos committed:
Revert "Cleanup rabbits remainings"

This reverts commit daf4e1f84cb67ab74c3fea5e4899a45757ec2dc7.

pamarcos committed:
Update rabbitmq docker image to latest: 4.2.1

pamarcos committed:
Increase flaky re-runs to 5000

pamarcos commented:
> Also need to revert this one in the future: #92154

I've already merged in this PR master and reverted that one

pamarcos committed:
Run several times flaky check

pamarcos committed:
Fix integration test flaky check label filter

pamarcos committed:
Increase timeouts for pulling docker image: 300 -> 600

Also increase timeout for test_storage_rabbitmq.

Example where the following happened:

https://s3.amazonaws.com/clickhouse-test-reports/json.html?REF=25.9&sha=7737bba9345896fee0e3805ee54bf8d8b51d34d7&name_0=ReleaseBranchCI&name_1=Integration%20tests%20%28amd_asan%2C%20old%20analyzer%2C%205%2F6%29&name_1=Integration%20tests%20%28amd_asan%2C%20old%20analyzer%2C%205%2F6%29

Timeline Breakdown:

  13:18:13  Test starts (cluster setup begins)
            ↓ (300 seconds)
  13:23:13  1st docker compose pull timeout (DOCKER_CLIENT_TIMEOUT)
            → Retry initiated after ~2 seconds
            ↓ (300 seconds)
  13:28:15  2nd docker compose pull timeout (DOCKER_CLIENT_TIMEOUT)
            → Retry initiated after ~2 seconds
            ↓ (298 seconds - almost made it!)
  13:33:13  PYTEST TIMEOUT KILLS EVERYTHING (900s total elapsed)

pamarcos committed:
Bring completely down the RabbitMQ instance and start from scratch

For some reason, stop_app/start_app sometimes get stuck.
Let's be pragmatic and completely kill the instance and bring a new
one instead.

pamarcos committed:
Add rabbitmq_management plugin

pamarcos committed:
Remove RabbitMQMonitor and use statistics from management plugin instead

pamarcos committed:
Ignore errors in case stats cannot be retrieved

pamarcos committed:
Make the RabbitMQ stats human-friendly

pamarcos committed:
Revert repeat count
